graveler/committed/tree/tree.go

```diff
@@ -1,21 +1,42 @@
 package tree
 
 //go:generate mockgen -source=tree.go -destination=mock/tree.go -package=mock
+//go:generate protoc --proto_path=. --go_out=. --go_opt=paths=source_relative tree.proto
 
 import (
+	"github.com/gogo/protobuf/proto"
 	"github.com/treeverse/lakefs/graveler"
 	"github.com/treeverse/lakefs/graveler/committed"
 )
 
-// part is the basic building stone of a tree
-// nolint: structcheck, unused
+// Part is the basic building stone of a tree
 type Part struct {
-	Name   committed.ID
-	MaxKey graveler.Key
+	ID                committed.ID
+	MinKey            committed.Key
+	MaxKey            committed.Key
+	EstimatedSize     int
+	ReachedBrakePoint bool
+}
+
+func MarshalPart(part Part) ([]byte, error) {
+	return proto.Marshal(&PartData{
+		MinKey:        part.MinKey,
+		MaxKey:        part.MaxKey,
+		EstimatedSize: int64(part.EstimatedSize),
+		BreakOnHash:   part.ReachedBrakePoint,
```

Comment:
@arielshaqed I started by using `PartData`, the problem is that the generated struct by Protoc contains a lock, which blocked me from passing it by value.

Reply:
Yuck, that's terrible!  But it gets worse, see https://stackoverflow.com/questions/64183794/why-do-the-go-generated-protobuf-files-contain-mutex-locks: it *doesn't* contain `mutex.Lock`, it contains a `[0]mutex.Lock` that relies on what They euphemistically call a "go vet feature" (more like a bug) to prevent copying.  And the reason you should prevent copying is that _it would be **immoral**_: of all the Go `struct`s in the world, you should not copy protobufs because that would only be a shallow copy.

But anyway: could we just get rid of `Part` and have _only_ `PartData`?  And hold it as a pointer because "_something something_ Go is better than K&R C and perl4"?

Reply:
Not sure we want to.
Part has ID as a member and part data does not,
and Doing that will force us not to pass Part as value in functions.